x-pack/plugins/enterprise_search/public/applications/app_search/components/result_field_value/result_field_value.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+import classNames from 'classnames';
+
+import { Raw, Snippet } from '../../types';
+
+import './result_field_value.scss';
+
+const isNotNumeric = (raw: string | number): boolean => {
+  if (typeof raw === 'number') return false;
+  return isNaN(parseFloat(raw));
+};
+
+const getRawArrayDisplay = (rawArray: Array<string | number>): string => {
+  return `[${rawArray.map((raw) => (isNotNumeric(raw) ? `"${raw}"` : raw)).join(', ')}]`;
+};
+
+const parseHighlights = (highlight: string): string => {
+  return highlight.replace(
+    /<em>(.+?)<\/em>/gi,
+    '<em class="enterpriseSearchResultHighlight" data-highlight="$1">$1</em>'
+  );
+};
+
+const isFieldValueEmpty = (type?: string, raw?: Raw, snippet?: Snippet) => {
+  const isNumber = type === 'number';
+  if (isNumber) {
+    return raw === null;
+  }
+
+  return !snippet && !raw;
+};
+
+interface Props {
+  raw?: Raw;
+  snippet?: Snippet;
+  type?: string;
+  className?: string;
+}
+
+export const ResultFieldValue: React.FC<Props> = ({ snippet, raw, type, className }) => {
+  const isEmpty = isFieldValueEmpty(type, raw, snippet);
+  const fieldValueClassName = classNames(
+    'enterpriseSearchResultFieldValue',
+    {
+      'enterpriseSearchDataType--empty': isEmpty,
+      [`enterpriseSearchDataType--${type}`]: !!type,
+    },
+    className
+  );
+  const fieldValueNode = snippet ? (
+    <div
+      className="enterpriseSearchResultSnippet"
+      /*
+       * Justification for dangerouslySetInnerHTML:
```

Comment:
I added this as well, since it seemed to be a pattern in Kibana, to disable this check and add a reason statement above.